packages/orbit-components/src/ChoiceTile/__tests__/index.test.tsx

```diff
@@ -0,0 +1,97 @@
+import * as React from "react";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+
+import * as mq from "../../hooks/useMediaQuery";
+import ChoiceTile from "..";
+
+describe("ChoiceTile", () => {
+  it("renders with correct properties", () => {
+    render(
+      <ChoiceTile title="title" description="description" label="label" icon="icon" type="radio">
+        Custom content
+      </ChoiceTile>,
+    );
+    expect(screen.getByText("title")).toBeInTheDocument();
+    expect(screen.getByText("description")).toBeInTheDocument();
+    expect(screen.getByText("label")).toBeInTheDocument();
+    expect(screen.getByText("icon")).toBeInTheDocument();
+    expect(screen.getByText("Custom content")).toBeInTheDocument();
+    expect(screen.getByRole("radio")).toBeInTheDocument();
+  });
+
+  it("can be checkbox", () => {
+    render(<ChoiceTile type="checkbox" />);
+    expect(screen.getByRole("checkbox")).toBeInTheDocument();
+  });
+
+  it("can be selected", () => {
+    render(<ChoiceTile type="radio" selected />);
+    expect(screen.getByRole("radio")).toBeChecked();
+  });
+
+  it("calls onClick", () => {
+    const onClick = jest.fn();
+
+    render(<ChoiceTile dataTest="choice-tile" type="radio" onClick={onClick} />);
+    userEvent.click(screen.getByTestId("choice-tile"));
+
+    expect(onClick).toHaveBeenCalled();
+  });
+
+  it("can be inline", () => {
+    const onClick = jest.fn();
+    const mqMock = jest.spyOn(mq, "default").mockImplementationOnce(() => {
```

Comment:
instead of spy, use the mock of `useMediaQuery`, something like [here](https://github.com/kiwicom/orbit/blob/master/packages/orbit-components/src/Pagination/__tests__/index.test.tsx)

Reply:
There are a few things I don't particularly like on that example. We have to mock and restore mock for every test (even if viewport is irrelevant for what's being tested) and by defining the type to `any` we can mock it to be anything really, not following types. Is there any major advantage in that approach that I am not seeing?

Reply:
you are already restoring it for almost all tests. afterEach can help to restore it in one function, instead of calling mockRestore at the end of each assertion. 

It just looks less verbose, there is a default exported hook and by mocking the module you do not need to spy on default. It's up to you

Reply:
With the new markup implemented (following the suggestion in the other thread), I have removed theses tests, as they were relying in the Description wrapper.

Therefore, I had to rethink the tests and came up with an approach that I don't particularly like. I implemented that on a separate commit, so that it is easier to review, and eventually revert, if we decide we don't like it. I would still like to have the logic on the `inline` prop tested. But maybe only Loki tests are enough for this, no?

Reply:
Yes, Loki testing might be enough ;)